devtools/src/object-explorer.js

```diff
@@ -1,7 +1,62 @@
 import {PolymerElement} from '../deps/@polymer/polymer/polymer-element.js';
 import {html} from '../deps/@polymer/polymer/lib/utils/html-tag.js';
 
-class ObjectExplorer extends PolymerElement {
+/**
+ * ..:: Read Before Modifying ::..
+ * 
+ * ObjectExplorer is implemented to work well within an IronList,
+ * which efficiently displays large collections by recycling DOM.
+ * 
+ * The above constraint means that we CANNOT do following:
+ *   - Call functions from the template.
+ *     E.g. <span>[[_helperFunction(item)]]</span>
+ *   - Store information on DOM elements.
+ *     E.g. information about state of expansion needs to be in the data model.
+ * 
+ * ..:: Read Before Using ::..
+ * 
+ * There are 2 parallel APIs to the ObjectExplorer.
+ * 
+ * 1) When using as a standalone custom element:
+ *   
+ *   Pass the object to explore through an 'object' attribute:
+ *     <object-explorer object="[[objectToExplore]]"></object-explorer>
+ * 
+ *   Trigger searching through a 'find' attribute:
+ *     <object-explorer object="[[objectToExplore]]" find="[[searchPhrase]]">
+ *     or:
+ *     this.$.explorer.find = searchPhrase;
+ * 
+ * 2) When using inside an IronList:
+ * 
+ *   There is more data in your list than actual DOM elements, so you need to
+ *   prepare the underlying ObjectExplorer data structure yourself. You co do it
+ *   with a ObjectExplorer.prepareData static method and pass the result into
+ *   a 'data' attribute of the object-explorer.
+ * 
+ *   this.items = objects.map(obj => ObjectExplorer.prepareData(obj));
+ * 
+ *   <iron-list items="[[items]]">
+ *     <object-explorer data="[[item]]"></object-explorer>
+ *   </iron-list>
+ * 
+ *   To perform a search, we need to manually trigger search change in the underlying
+ *   data model, as there is more data than ObjectExplorer DOM elements. We do it with
+ *   ObjectExplorer.find(data, phrase) method.
+ *     
+ *     this.items.forEach(item => ObjectExplorer.find(item, searchPhrase));
+ * 
+ *   As the ObjectExplorer DOM elements have not been notified of the underlying data
+ *   changes, we need to also notify them manually. We can do it by setting a 'find'
+ *   attribute.
+ * 
+ *     for (const explorer of this.shadowRoot.querySelectorAll('iron-list object-explorer')) {
+ *       explorer.find = phrase;
+ *     }
+ * 
+ *   When in doubt, talk to piotrswigon@github / piotrs@google.
+ */
```

Comment:
Nice!

Reply:
Thank... this small class embodies many hours of struggle and irritation, though it finally works.